### PR TITLE
backend/fix: added merchantOpCityId to stations list api

### PR DESCRIPTION
**Pull request by VD321**
## Type of Change

- [X] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/frfs-ticket.yaml [hunk 1]
```diff
@@ -65,9 +66,10 @@ Station:
       kvFunction: findAllWithKV
       where:
         and: [merchantOperatingCityId, vehicleType]
-    getTicketPlacesByVehicleType:
+    getTicketPlacesByVehicleTypeAndMOCityId:
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/frfs-ticket.yaml:69]: revert this change

File: Backend/app/rider-platform/rider-app/Main/src-read-only/Storage/Queries/Station.hs [hunk 2]
```diff
@@ -34,19 +35,22 @@ findByStationCode code = do
...
     [ Se.And
-        [ Se.Is Beam.merchantOperatingCityId $ Se.Eq (Kernel.Types.Id.getId <$> merchantOperatingCityId),
+        [ Se.Is Beam.merchantOperatingCityId $ Se.Eq merchantOperatingCityId,
           Se.Is Beam.vehicleType $ Se.Eq vehicleType
         ]
     ]
 
-getTicketPlacesByVehicleType :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Domain.Types.Station.FRFSVehicleType -> m ([Domain.Types.Station.Station])
-getTicketPlacesByVehicleType vehicleType = do
+getTicketPlacesByVehicleTypeAndMOCityId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Domain.Types.Station.FRFSVehicleType -> Kernel.Types.Id.Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity -> m ([Domain.Types.Station.Station])
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src-read-only/Storage/Queries/Station.hs:47]: revert

**Review by RitikaHanish — CHANGES_REQUESTED** [review]

**Review by RitikaHanish — CHANGES_REQUESTED** [review]
Handle sending correct city in every beckn FRFS context.

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnConfirm.hs [hunk 3]
```diff
@@ -157,6 +159,16 @@ callBPPCancel :: DFRFSTicketBooking.FRFSTicketBooking -> BecknConfig -> Environm
 callBPPCancel booking bapConfig = do
   fork "FRFS Cancel Req" $ do
     providerUrl <- booking.bppSubscriberUrl & parseBaseUrl & fromMaybeM (InvalidRequest "Invalid provider url")
-    bknCancelReq <- ACL.buildCancelReq booking bapConfig Utils.BppData {bppId = booking.bppSubscriberId, bppUri = booking.bppSubscriberUrl}
+    merchantOperatingCity <- getMerchantOperatingCityFromBooking booking
+    bknCancelReq <- ACL.buildCancelReq booking bapConfig Utils.BppData {bppId = booking.bppSubscriberId, bppUri = booking.bppSubscriberUrl} merchantOperatingCity.city
     logDebug $ "FRFS SearchReq " <> (encodeToText bknCancelReq)
     void $ CallBPP.cancel providerUrl bknCancelReq
+
+getMerchantOperatingCityFromBooking :: DFRFSTicketBooking.FRFSTicketBooking -> Environment.Flow DMOC.MerchantOperatingCity
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnConfirm.hs:167]: Move this to Utils

Comment (VD321) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnConfirm.hs:167]: Done

**Review by RitikaHanish — APPROVED** [review]

**Pull request merged**